### PR TITLE
Add documentation and tests to min and max functions

### PR DESCRIPTION
**Pull request by pineapplemachine**

### CODE REVIEW

File: src/functions/max.js [hunk 1]
```diff
@@ -1,20 +1,44 @@
...
+        expects: (`
+            The function expects a sequence as input an an optional
+            [relational function].
+            If no relational function was passed, the default relational
+            function \`(a, b) => (a < b)\` is used.
+        `),
+        returns: (`
+            The function returns the maximum element in the sequence according
+            to the relational function. Functionally, this would be the same
+            as [stable-sorting](stable sort) the sequence according to that
```
Comment (pineapplemachine) [inline src/functions/max.js:19]: Not yet a glossary term, but it will be

File: src/functions/max.js [hunk 2]
```diff
@@ -1,20 +1,44 @@
...
+        examples: [
+            "basicUsage", "withRelationalFunction"
+        ],
+    },
     attachSequence: true,
     async: true,
     arguments: {
         unordered: {
-            functions: 1,
+            functions: "?",
```
Comment (pineapplemachine) [inline src/functions/max.js:33]: Bugfix

File: src/functions/max.js [hunk 3]
```diff
@@ -1,20 +1,44 @@
...
             sequences: 1,
-            allowIterables: true
-        }
+            allowIterables: true,
+        },
     },
     implementation: (relate, source) => {
+        NotBoundedError.enforce(source, {
+            message: "Failed to find a maximum element"
+        });
```
Comment (pineapplemachine) [inline src/functions/max.js:41]: Also bugfix

File: src/functions/max.js (hunk 1 above)
Comment (jpsullivan) [reply src/functions/max.js:19]: Created an issue for this so it isn't forgotten -- I would totally forget to do this.
https://github.com/pineapplemachine/higher/pull/66/files#r127577462

**Review by jpsullivan — APPROVED** [review]

**Pull request merged**